api/handlers_test.go

```diff
@@ -86,6 +125,51 @@ func TestDatasetGet(t *testing.T) {
 	assertStatusCode(t, "invalid dsref", actualStatusCode, 400)
 }
 
+func TestUnpackHandler(t *testing.T) {
+	buf := new(bytes.Buffer)
+	zw := zip.NewWriter(buf)
+
+	text := []byte(`{ "meta": { "title": "hello world!" }}`)
+	filename := "meta.json"
+	f, err := zw.Create(filename)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = f.Write(text)
+	if err != nil {
+		t.Fatal(err)
+	}
+	zw.Close()
+
+	rr := bytes.NewReader(buf.Bytes())
+
+	r := httptest.NewRequest("POST", "/unpack", rr)
+	w := httptest.NewRecorder()
+
+	hf := UnpackHandler("/unpack")
+	hf(w, r)
+	if w.Result().StatusCode != 200 {
+		t.Errorf("%s", w.Body.String())
+		t.Fatal(fmt.Errorf("expected unpack handler to return with 200 status code, returned with %d", w.Result().StatusCode))
+	}
+
+	r = httptest.NewRequest("GET", "/unpack", nil)
+	w = httptest.NewRecorder()
+	hf(w, r)
+	if w.Result().StatusCode != 404 {
+		t.Errorf("%s", w.Body.String())
+		t.Fatal(fmt.Errorf("expected call to unpack handler with GET method to return status code 404, got %d", w.Result().StatusCode))
+	}
+
+	r = httptest.NewRequest("POST", "/unpack", nil)
+	w = httptest.NewRecorder()
+	hf(w, r)
+	if w.Result().StatusCode != 500 {
+		t.Errorf("%s", w.Body.String())
+		t.Fatal(fmt.Errorf("expected call to unpack handler with GET method to return status code 404, got %d", w.Result().StatusCode))
```

Comment:
The error message says 404, but the condition is checking 500. They should match.

Reply:
EAGLE EYESSSSS ty